File: 1. 단계별로_풀어보기/1~9/9. 약수, 배수와 소수 1단계/2. 2501.py
import math, sys

input = sys.stdin.readline
output = sys.stdout.write

N, K = map(int, input().split())

maxIter = math.floor(math.sqrt(N))

factorArr = []

for i in range(1, maxIter + 1):
  if N % i == 0:
    factorArr.append(i)
    
    div = N // i
    if div != i:
      factorArr.append(div)

factorArr.sort()

if len(factorArr) < K:
  output(f"0")
else:
  output(f"{factorArr[K - 1]}")

File: 1. 단계별로_풀어보기/1~9/9. 약수, 배수와 소수 1단계/2. 2501.js
// 입력
const fs = require('fs')
const inputArr = fs.readFileSync(0, 'utf-8').toString().trim().split('\n')

// 입력 파싱
const [N, K] = inputArr[0].split(' ').map(Number)

// 약수를 담을 배열
const factorArr = []

// 약수들을 찾아서 배열에 삽입

for (let i = 1; i <= N; i++) {
  if (N % i === 0) {
    factorArr.push(i)
  }
}

if (factorArr.length < K) {
  process.stdout.write('0')
} else {
  process.stdout.write(`${factorArr[K - 1]}`)
}
